docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-planning-problem-con.adoc

```diff
@@ -0,0 +1,53 @@
+
+The ability to achieve these goals relies on the number of resources available, such as:
+
+* The number of people
+* Amount of time
+* Budget
+* Physical assets, for example, machinery, vehicles, computers, buildings, etc
+
+Specific constraints related to these resources must also be taken into account, such as the number of hours a person works, their ability to use certain machines, or compatibility between pieces of equipment. 
+ 
+The Business Optimizer helps Java programmers solve constraint satisfaction problems efficiently. Under the hood, it combines optimization heuristics and metaheuristics with efficient score calculation.
+
+
+[id='optimizer-planning-problem-search-space-con']
+== A Planning Problem Has a Huge Search Space
+
+A planning problem has a number of solutions.
+
+There are several categories of solutions:
+
+Possible solution::
+A possible solution is any solution, whether or not it breaks any number of constraints. Planning problems tend to have an incredibly large number of possible solutions. Many of those solutions are worthless.
+
+Feasible solution::
+A feasible solution is a solution that does not break any (negative) hard constraints. The number of feasible solutions tends to be relative to the number of possible solutions. Sometimes there are no feasible solutions. Every feasible solution is a possible solution.
+
+Optimal solution::
+An optimal solution is a solution with the highest score. Planning problems tend to have 1 or a few optimal solutions. There is always at least 1 optimal solution, even in the case that there are no feasible solutions and the optimal solution is not feasible.
+
+Best solution found::
+The best solution found is the solution with the highest score found by an implementation in a given amount of time. The best solution found is likely to be feasible and, given enough time, it's an optimal solution.
+
+Counterintuitively, the number of possible solutions is huge (if calculated correctly), even with a small data set.
+
+In the examples provided in the `planner-engine` distribution folder, most instances have a significantly higher number of possible solutions than the minimal number of atoms in the known universe (10^80). As there is no guaranteed way to find the optimal solution, any implementation is forced to evaluate at least a subset of all those possible solutions.
+
+The Business Optimizer supports several optimization algorithms to efficiently wade through that incredibly large number of possible solutions.
+
+Depending on the use case, some optimization algorithms perform better than others, but it is impossible to tell in advance. Using the Business Optimizer, you can switch the optimization algorithm by changing the solver configuration in a few lines of XML or code.
+
+
+include::optimizer-NP-complete-vs-NP-hard-con.adoc[leveloffset=+1]
+
+include::optimizer-hard-vs-soft-constraints-con.adoc[leveloffset=+1]
```

Comment:
So if these two includes must be part of the "A planning problem has a huge search space" module for it to be complete (i.e., it truly can't and won't ever be able to be reused without these two other dependent modules included in it) then this is fine. But if that's not the case, then move these includes to the assembly, following this module's include (which again should be a separate include, rather than nested in "What is a planning problem?". Probably seems tedious, but it's all about creating building blocks to be as reusable as possible. As you know.